x-pack/plugins/ml/public/util/field_types_utils.js

```diff
@@ -39,3 +41,48 @@ export function kbnTypeToMLJobType(field) {
 
   return type;
 }
+
+export const mlJobTypeAriaLabels = {
+  BOOLEAN: i18n.translate('xpack.ml.fieldTypeIcon.booleanTypeAriaLabel', {
+    defaultMessage: 'boolean type',
+  }),
+  DATE: i18n.translate('xpack.ml.fieldTypeIcon.dateTypeAriaLabel', {
+    defaultMessage: 'date type',
+  }),
+  GEO_POINT: i18n.translate('xpack.ml.fieldTypeIcon.geo_pointTypeAriaLabel', {
+    defaultMessage: 'geo point type',
+  }),
+  IP: i18n.translate('xpack.ml.fieldTypeIcon.ipTypeAriaLabel', {
+    defaultMessage: 'ip type',
+  }),
+  KEYWORD: i18n.translate('xpack.ml.fieldTypeIcon.keywordTypeAriaLabel', {
+    defaultMessage: 'keyword type',
+  }),
+  NUMBER: i18n.translate('xpack.ml.fieldTypeIcon.numberTypeAriaLabel', {
+    defaultMessage: 'number type',
+  }),
+  TEXT: i18n.translate('xpack.ml.fieldTypeIcon.textTypeAriaLabel', {
+    defaultMessage: 'text type',
+  }),
+  UNKNOWN: i18n.translate('xpack.ml.fieldTypeIcon.unknownTypeAriaLabel', {
+    defaultMessage: 'unknown type',
+  }),
+};
+
+export const getMLJobTypeAriaLabel = (keyword) => {
+
+  let requestedFieldType = Object.keys(ML_JOB_FIELD_TYPES)
+    .map(constantName => ({
+      constantName,
+      matched: !!(ML_JOB_FIELD_TYPES[constantName] === keyword),
+    }))
+    .filter(item => item.matched);
+
+  if (!requestedFieldType.length) {
+    return null;
+  } else {
+    requestedFieldType = requestedFieldType[0];
+  }
+
+  return mlJobTypeAriaLabels[requestedFieldType.constantName] || null;
```

Comment:
this function seems overly complicated. 
the variable names `keyword` and `constantName` don't make sense to me. the parameter being passed in would make more sense if it were called `type`.
`keyword` is a value that could be passed in, and so reading this i would expect only keyword types to be passed in.

the function could be made a lot simpler IMO:
```
export function getMLJobTypeAriaLabel(type) {
  const t = Object.keys(ML_JOB_FIELD_TYPES).find(k => (ML_JOB_FIELD_TYPES[k] === type));
  return mlJobTypeAriaLabels[t] || null;
}
```

Reply:
@jgowdyelastic , agreed, thanks a lot!